pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy

```diff
@@ -319,15 +320,30 @@ public class Utils {
         String toEval = s ?: ""
         if (!toEval.startsWith('"') || !toEval.endsWith('"')) {
             if (toEval.indexOf('\n') == -1) {
-                toEval = '"' + toEval + '"';
+                toEval = '"' + escapeForEval(toEval) + '"';
             } else {
-                toEval = '"""' + toEval + '"""';
+                toEval = '"""' + escapeForEval(toEval) + '"""';
             }
         }
 
         return toEval
     }
 
+    static String escapeForEval(String s) {
+        s = StringUtils.replace(s, '\\\\', '\\\\')
```

Comment:
Amusingly, it does do something! Basically it replaces `\\` with `\\\\`. It just looks funny here.